api/specs.proto

```diff
@@ -84,7 +84,7 @@ message ServiceSpec {
 		// Aliases specifies a list of discoverable alternate names for the service on this Target.
 		repeated string aliases = 2;
 	}
-	repeated NetworkAttachmentConfig networks = 7;
```

Comment:
@stevvooe The one problem with removing the embedded `NetworkAttachmentConfig` is that the type encoding of state from old version cannot be ready any more because I think the type encoding of the non-embedded version of this type will differ from the embedded version.